accepted/00000-rbac.md

```diff
@@ -0,0 +1,285 @@
+
+- **`access_mode`**: Defines the access level for each namespace as either "View" (R) or "Modify" (W). Many namespaces will have separate entries for each mode, with different endpoints for each purpose.
+
+  For example, `clm.project.details` **[R]** allows viewing a project's details, where `clm.project.details` **[W]** allows modifying them.
+
+- **`description`**: A clear description of what a namespace grants access to.
+
+- **`scope`**: Indicates whether the endpoint is accessible through the web UI (including internal API calls) or the public API.
+
+#### Organization of namespaces
+
+Organizing namespaces in a hierarchical structure with distinct access modes simplifies management and modification of access rules for administrators across users or groups.
+
+### Access group
+
+An **access group** is a collection of access rules that can be assigned to users, allowing administrators to manage permissions for multiple users at once.
+Access groups can be either predefined or created by administrators.
+Predefined groups cannot be modified by the users and serve as replacements for existing user roles on the Uyuni server.
+
+- **`label`**: The label of the access group.
+
+- **`description`**: The purpose or scope of the access group.
+
+- **`org_id`**: The ID of the organization to which the group belongs. This field will be `null` for predefined groups.
+
+#### Existing roles
+
+In Uyuni, two superuser roles grant unrestricted access to system resources:
+- **Uyuni Administrator:** Grants access to every resource across the Uyuni system.
+- **Organization Administrator:** Grants access to all resources within the user's organization.
+
+Beyond feature access, these roles also control data visibility. For instance, an Uyuni Administrator can view all clients registered in Uyuni, regardless of the organization each client belongs to. Similarly, an Organization Administrator can view every client within their organization, even if managed by a different user. In contrast, regular users can only view the clients they manage.
+
+Due to these distinctions, the logic around superuser roles will remain in place. Additionally, these roles will have access to all namespaces in the new RBAC implementation.
+
+The other existing roles in Uyuni are as follows:
+
+- Activation Key Administrator
+- Configuration Administrator
+- Image Administrator
+- Channel Administrator
+- System Group Administrator
+- Read-only API User
+
+Each of these roles grants access to different features in Uyuni. This static access control is enforced individually within each feature. Stripping these control checks throughout the system requires significant effort. Therefore, in the initial phase, the logic of these static roles will coexist with their RBAC counterparts, but will be bypassed by assigning them to all existing and future users. This will effectively make RBAC the exclusive access control mechanism.
```

Comment:
This is not clear to me. Yes, we will keep the existing roles so we don't break the users setup when the migrate to 5.1.  But wow are we planning to get rid of them? Will we allow administrators to still play with old roles ? Having new and old permissions setup could be pretty confusing for the admin to understand.

Reply:
There are 2 roles that needs to stay for now, which are `satadmin` and `orgadmin`.
Other roles should be replaced with new user groups in the new system.
Internally, we may need to keep the old roles because of struts ACL configurations.